[함헌규] 스프린트 미션 1&2

heonq opened the pull request:
## 요구사항

### 기본

#### 스프린트 미션 1

- [x] PC사이즈만 고려해 주어진 디자인을 구현합니다.
- [x] HTML, CSS 파일을 Netlify로 배포해 주세요.
- [x] “로그인”버튼 클릭 시 로그인 페이지(‘/login)로 이동합니다.
- [x] “구경하러가기”버튼 클릭 시(’/items’)로 이동합니다.
- [x] 아래로 스크롤 해도 “판다 마켓” 로고와 “로그인” 버튼이 있는 상단 네비게이션 바가 최상단에 고정되게 해주세요.
- [x] 화면의 너비가 1920px 보다 작아질 때, “판다마켓” 로고의 왼쪽 여백 200px, “로그인" 버튼의 오른쪽 여백 200px이 유지되고, 화면의 너비가 작아질수록 두 요소간 거리가 가까워지도록 설정합니다.
- [x] 화면의 너비가 1920px 이상이면 하늘색 배경색은 너비를 꽉 채우도록 채워지고, 내부 요소들의 위치는 고정되고, 여백만 커지도록 합니다.
- [x] 화면의 너비가 1920px 이상이면 내부에 있는 요소간 동일한 간격을 유지하며 가운데 정렬해야 합니다.
- [x] 클릭으로 기능이 동작해야 하는 경우, 사용자가 클릭할 수 있는 요소임을 알 수 있도록 CSS 속성 cursor: pointer 로 설정합니다.
- [x] 화면의 너비가 1920px 보다 작아질 때 “codeit-2024”의 왼쪽 여백 200px과 SNS 아이콘들의 오른쪽 여백 200px을 유지하면서 가운데 있는 “Privacy Policy”, “FAQ” 요소와 각각 동일한 간격을 유지하며 가까워져야 합니다.
- [x] Privacy Policy는 클릭 시 Privacy 페이지(‘/privacy’) 이동합니다. (빈 페이지)
- [x] FAQ는 클릭 시 FAQ 페이지(‘/faq’)로 이동합니다. (빈페이지)
- [x] 페이스북, 트위터, 유튜브, 인스타그램 아이콘을 클릭 시 각각의 홈페이지로 새로운 창이 열리면서 이동합니다.

#### 스프린트 미션 2

- [x] PC사이즈만 고려해 주어진 디자인을 구현합니다.
- [x] HTML, CSS 파일을 Netlify로 배포해 주세요.

##### 로그인 페이지, 회원가입 페이지 공통

- [x] “판다마켓" 로고 클릭 시 루트 페이지(“/”)로 이동합니다.
- [x] 로그인 페이지, 회원가입 페이지 모두 로고 위 상단 여백이 동일합니다.
- [x] SNS 아이콘들은 클릭 시 각각 “https://www.google.com/”, “https://www.kakaocorp.com/page/” 으로 이동합니다.
- [x] input 요소에 focus in 일 때, 테두리 색상은 ##3692FF입니다.
- [x] input 요소에 focus out 일 때, 테두리는 없습니다.

##### 로그인 페이지

- [x] “회원가입”버튼 클릭 시 “/signup” 페이지로 이동합니다.

##### 회원가입 페이지

- [x] “로그인”버튼 클릭 시 “/login” 페이지로 이동합니다.

### 심화

#### 스프린트 미션 1

- [x] reset.css를 설정해주세요.
- [x] 사용자의 브라우저 설정에 따라 기본 폰트 크기 설정이 변화함에 따라서 페이지의 요소 간 간격, 요소의 크기, font-size 등 모든 크기와 관련된 값이 크고 작아지도록 설정해주세요.

#### 스프린트 미션 2

- [x] 비밀번호, 비밀번호 확인 input 요소 오른쪽에 비밀번호를 확인할 수 있는 눈 모양 아이콘을 추가합니다.
- [x] palette에 있는 color값들을 css 변수로 등록해서 사용합니다.
- [x] 구글 애널리틱스로 방문자 수 확인하기할 수 있도록 설정합니다.

## 주요 변경사항

### 스프린트 미션 1
- 랜딩 페이지 제작

### 스프린트 미션 2
- 로그인 페이지 제작
- 회원가입 페이지 제작

## 스크린샷
- 배포 링크 https://adorable-phoenix-7e2244.netlify.app

### 랜딩 페이지
![스크린샷 2024-09-23 오후 1 37 48](https://github.com/user-attachments/assets/b80394ad-cbba-449f-b656-aef5885ff7dd)


### 로그인 페이지
![스크린샷 2024-09-23 오후 5 05 36](https://github.com/user-attachments/assets/d11a2dc1-4ce9-4c54-acf1-e0370b092aa0)


### 회원가입 페이지
![스크린샷 2024-09-23 오후 5 05 41](https://github.com/user-attachments/assets/cef7a9df-94d1-45c8-abb8-ad11df7d16c9)



## 멘토에게

- 시맨틱 태그를 올바르게 사용했는지 사용한 div나 span 태그 중 시맨틱 태그로 대체할 수 있는 경우가 있는지 궁금합니다.
- pages 폴더를 만들고 그 안에 각 페이지 별로 폴더를 만들어 index.html,styles.css를 작성했는데 이러한 폴더 구조가 적절한지 궁금합니다.

sprint-edu committed:
docs: update branch 구조 in readme

sprint-edu committed:
docs: update README.md

sprint-edu committed:
docs: update readme

프론트엔드/백엔드 레포 분리

sprint-edu committed:
docs: update readme

프론트엔드/백엔드 레포 분리

sprint-edu committed:
feat: add: auto-labeling and assignee for PRs

heonq committed:
feat: 스프린트 미션 2 완료

heonq committed:
style: html 태그의 클래스명 변경

heonq committed:
style: 로그인 페이지 내 태그의 클래스명 변경

heonq committed:
style: 컬러 값에 CSS 변수 생성하여 적용

heonq committed:
style: 로그인,계정생성 페이지 스타일 변경

heonq committed:
docs: pull-request-template 수정

heonq committed:
docs: pull-request-template 수정

heonq committed:
style: 로그인,계정생성 페이지 스타일 변경

heonq committed:
docs: pull-request-template 파일 롤백

heonq committed:
style: 로그인,회원가입 페이지 스타일 변경

- main태그에 폰트컬러를 적용하고 개별 요소에 적용한 폰트 컬러 속성을 제거

orlein commented on .gitignore:
```diff
@@ -0,0 +1 @@
+.DS_Store
```
애플에서 작업하시는분들이 놓치기 쉬운 부분을 잘 잡아주셨네요. 좋습니다.
https://www.toptal.com/developers/gitignore/ 여기서 개발 환경 이름들을 입력하고 `.gitignore` 파일 내용을 그대로 복붙하시면 편리합니다.

heonq replied:
유용한 사이트네요! 바로 적용했습니다!

orlein commented on index.html:
```diff
@@ -0,0 +1,144 @@
+            />
+            <div class="main-page-text align-left">
+              <span class="sub-title">Hot item</span>
+              <h1 class="main-title">
+                인기 상품을<br />
+                확인해 보세요
+              </h1>
```
`인기 상품을` `확인해 보세요` 는 서로 떨어져있는 것이 맞고, 컴포넌트화를 시키더라도 떨어지겠지만, 자연스럽게 word-break으로 떨어지게끔 한번 컴포넌트를 꾸며보세요. 
나중에 프론트엔드 작업을 할 때, `1) 원래 문구가 떨어져있음 2) 원래 붙어있는데 내 맘대로 떨어뜨릴 수 있음` 이 두 가지는 꽤 다른 결과를 냅니다.

orlein commented on index.html:
```diff
@@ -0,0 +1,144 @@
+            <a href="./pages/privacy/index.html">Privacy Policy</a>
+            <a href="./pages/faq/index.html">FAQ</a>
+          </div>
+          <div class="social-media-link">
+            <a target="_blank" href="https://www.facebook.com"
+              ><img src="./images/ic_facebook.png" alt="페이스북 로고"
+            /></a>
+            <a target="_blank" href="https://x.com">
+              <img src="./images/ic_twitter.png" alt="x(구 트위터) 로고" />
+            </a>
+            <a target="_blank" href="https://youtube.com">
+              <img src="./images/ic_youtube.png" alt="유튜브 로고" />
+            </a>
+            <a target="_blank" href="https://instagram.com">
+              <img src="./images/ic_instagram.png" alt="인스타그램 로고" />
+            </a>
+          </div>
```
로고 이미지에 alt를 넣어주신 점 너무 좋습니다.

orlein commented on pages/faq/index.html:
```diff
@@ -0,0 +1,19 @@
+      gtag("js", new Date());
+      gtag("config", "G-ZNH254KW2D");
+    </script>
+  </head>
+  <body></body>
+</html>
```
다른 index.html 페이지에서도 마찬가지로, 정말로 이렇게 `빈 페이지`를 로딩하게 되면 개발하는 사람이나 보는 사람이나 이 페이지의 정체를 찾기위해 약간의 수고를 해야합니다. 본인을 위해서라도 간단한 문구 (이 페이지의 경우에는 FAQ) 를 넣어주세요.
![image](https://github.com/user-attachments/assets/4f6e80a8-0005-4ac2-9db5-f37c1ea878d1)
일하다 보면 이렇게 title도 안보이는경우가 많아서 그렇습니다.

orlein commented on index.html:
```diff
@@ -0,0 +1,144 @@
+<!DOCTYPE html>
+<html lang="en">
```
사소합니다만 `lang` 은 한국어를 사용하고있으니 `ko` 를 사용해주세요.

orlein commented on pages/login/styles.css:
```diff
@@ -0,0 +1,144 @@
+:root {
+  --main-background-color: #fcfcfc;
+  --main-ivory-color: #f3f4f6;
+  --input-focus-border-color: #3692ff;
+  --main-grey-color: #9ca3af;
+  --main-font-color: #1f2937;
+  --social-login-bg-color: #e6f2ff;
+  --sign-up-link-font-color: #3182f6;
+  --password-visible-button-color: #4b5563;
+}
```
css variable로 컬러 표현하는것 정말 좋습니다. 혹여 나중에 디자이너분과 협업하실 때, `semantic color` 라는 개념을 얘기하신다면 그걸 css variable로 가져오면 더 좋습니다.

orlein commented on pages/signup/styles.css:
이 파일과 다른 파일이 서로 같은 내용을 공유하는것으로 보입니다. 되도록이면 같은 내용을 반복해야만 하는 상황을 피하시고, 정말로 꼭 그래야만 한다면 주석으로 그래야만 하는 당위성을 메모해두세요.
그렇게 같은 내용이 복사되었을 때, 나중에 그 내용이 변경될 경우 찾아바꾸기 기능으로 일일이 하나씩 다 바꿔주고 실수가 없는지 체크해야합니다. 이는 야근의 원인중에 가장 큰 원인입니다.

heonq replied:
클래스 선택자가 하나 다른 부분이 있었는데 이를 하나의 CSS파일로 모으고 중복된 CSS파일은 삭제했습니다!
감사합니다.

heonq committed:
style: 로그인,회원가입 페이지 스타일 변경

- 인풋창이 포커스 됐을 때 border가 아닌 outline을 적용

orlein commented on index.html:
```diff
@@ -0,0 +1,144 @@
+        <section class="banner bottom">
+          <div class="banner-content">
+            <div class="banner-text-container bottom">
+              <p class="banner-text bottom">
+                믿을 수 있는<br />
+                판다마켓 중고 거래
+              </p>
```
![image](https://github.com/user-attachments/assets/745fbbd4-b70a-4938-8a0f-a08dd79b92cd)
애플이나 마소같은 대기업도 종종 하는 실수입니다 ^^; 한번 다시 잡아보세요

orlein reviewed (commented):
불필요한 반복만 피한다면 전체적으로 훌륭한 결과물입니다. 아주 좋습니다.

heonq committed:
chore: .gitignore 파일 수정

- macOs 환경에 맞게 gitignore 설정

heonq committed:
style: HTML 파일 lang을 en->ko로 변경

heonq committed:
feat: 미완성 html 파일에 기본 내용 추가

heonq committed:
style: 로그인,회원가입 페이지에 적용할 CSS파일 변경

- 로그인,회원가입 페이지에 적용되는 각각의 CSS파일이 한개의 클래스 선택자를 제외하고 동일:
하나의 CSS파일에 나머지 CSS파일에 있던 클래스 선택자를 추가하고 두개의 페이지에서 동일한 CSS파일을 적용하도록 변경

heonq committed:
style: 랜딩페이지 태그 및 스타일 변경

- 텍스트의 줄을 나눌 때 <br/>태그를 사용하지 않고 word-break를 통해 줄을 나누도록 설정:
줄이 나눠지는 width를 계산하여 max-width로 설정

heonq committed:
style: 로그인,회원가입 페이지 태그 수정

- 구글,카카오 계정 로그인 이미지에 alt태그 추가

- 로고 부분의 태그를 section에서 div로 변경

heonq committed:
style: 랜딩페이지 스타일 변경

heonq committed:
style: 랜딩페이지 스타일 변경

miridih-jhkim06 commented on index.html:
```diff
@@ -0,0 +1,130 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Panda Market</title>
+    <link rel="styleSheet" href="reset.css" />
+    <link rel="styleSheet" href="styles.css" />
```
(사소) `styleSheet` > `stylesheet`

miridih-jhkim06 commented on index.html:
```diff
@@ -0,0 +1,130 @@
+      <header>
+        <div class="header-container">
+          <img class="logo" width="151" height="51" src="/images/logo.png" alt="판다마켓 로고" />
+          <a href="./pages/login/index.html"><button class="login-button">로그인</button></a>
```
<a>태그 내에 <button>태그를 사용하는것은 권장하지 않습니다.

첫 번째로는 접근성에 관한 문제입니다. 웹 접근성을 통해서 컨텐츠를 읽는 스크린리더기는 HTML태그를 기준으로 내용을 이해합니다. 이때 의미적인 내용이 충돌(<a>태그는 링크, <button>태그는 다른 스크립트에대한 실행)이 발생할 수 있어 버그를 유발할 수 있습니다.

두번째로는 기능에 대한 충돌입니다. 여기서는 버튼에 onClick과 같은 동작이 존재하지않지만, 이또한 버그를 유발할 수 있는 코드입니다.

수정한다면 다음과 같이 수정할 수 있을것 같아요.

```diff
- <a href="./pages/login/index.html"><button class="login-button">로그인</button></a>
+ <a href="./pages/login/index.html" class="login-button">로그인</a>
```

miridih-jhkim06 commented on styles.css:
```diff
@@ -0,0 +1,273 @@
+}
+
+html {
+  font-size: 62.5%;
```
html의 기본폰트사이즈를 줄이는 스펙이 있었을까요? html태그의 폰트사이즈를 낮게 설정하는건 가독성에 문제가 생길 수 있을것 같습니다.

heonq replied:
스프린트 미션 1 심화 요구사항 중 rem을 사용하라는 요구사항이 있었습니다! 이때 폰트 사이즈를 62.5%로 설정하면 1rem이 10px로 설정되어 rem을 계산하기 쉬워서 이렇게 설정했는데요! body태그에 font-size를 1.6rem으로 설정하는 식으로 폰트 사이즈를 다시 기본값으로 유지하는 방식으로 해도 괜찮을까요?

kimjong95 replied:
아하 rem값 계산의 직관성을 위해 줄여두었다는 말씀이신거군요?! 이렇게는 생각해본적이 없었는데 고민을 해봐야겠어요!

이러한 결정사항에는 트레이드오프가 존재할 것 같습니다. 
웹 접근성을 생각하는 대부분의 웹은 기본설정(100%)를 사용하기에 웹 기본설정기반인 스크린리더기에 의한 접근성에서는 좋지 않을 수 있습니다. 
반대로 이러한 방식으로사용하면 rem의 계산법이 조금더 수월할 수도 있겠네요.

miridih-jhkim06 commented on pages/login/index.html:
```diff
@@ -0,0 +1,59 @@
+  </head>
+  <body>
+    <main>
+      <div class="logo">
```
(사소) 의미가 있는 시멘틱 태그를 사용한다면 다른 태그를 이용할 수 있지 않을까요?!

heonq replied:
이부분은 header태그로 변경했습니다! 감사합니다!

miridih-jhkim06 reviewed (commented):
이외에는 다른 멘토님이 남겨주셔서 코드에서 리뷰할 내용은 더이상 없을것 같아요! 좋습니다!!

miridih-jhkim06 commented:
추가적으로 남겨주신 질문에 대한, 
1. 시맨틱 태그를 올바르게 사용했는지 사용한 div나 span 태그 중 시맨틱 태그로 대체할 수 있는 경우가 있는지 궁금합니다.
-> 이건 대부분 바르게 사용하신것 같습니다. 사소한 내용은 코드에 코멘트를 드렸어요.

2. pages 폴더를 만들고 그 안에 각 페이지 별로 폴더를 만들어 index.html,styles.css를 작성했는데 이러한 폴더 구조가 적절한지 궁금합니다.
페이지들을 폴더로 구분하고, 작성해주신것은 좋습니다. 다만 스펙상으로 페이지가 루트부터 /login, /signup, /faq 와같이 루트에서 1뎁스의 URL인데 pages 폴더내부에 넣으시면서 각각의 경로가 /pages/login, /pages/signup 등등으로 변환되었습니다. 이는 앞으로 구현하실 내용에서는 폴더구조를 통해서 코드의 가독성을 높힐 수 있으나 현재 요구사항과는 맞지 않는 모습을 보입니다.

또한 login페이지와 signup페이지가 같은 css파일을 참조하고있는데, 폴더구조(라우팅)에서는 해당 컨텍스트가 연결되어있지 않아보입니다.

폴더구조에 관한 고민은 아주 좋은것 같습니다. 실제로 팀단위 협업을하며 폴더구조와 잘 짜여진 컨벤션은 프로젝트의 품질을 높히는데 아주 도움이 됩니다!!

heonq committed:
style: 대소문자 수정

heonq committed:
feat: 랜딩페이지 중첩 태그 삭제

kimhyunmook committed:
mission

heonq committed:
style: 로그인,회원가입 페이지 태그 변경

heonq committed:
chore: 디렉토리 구조 변경

miridih-jhkim06 commented on assets/styles/login&signup.css:
```diff
@@ -0,0 +1,146 @@
+:root {
```
특별한 컨벤션이 없다면 파일이름에는 특수문자를 넣지 않는것이 좋을것 같습니다.
login(signIn)과 signup을 모두 표현하고 싶으시다면 sign.css 형태로 구분하시는것도 좋을 것 같습니다.

heonq committed:
chore: CSS파일명 변경

- 파일명에 있는 특수문자 제거

heonq committed:
style: 랜딩 페이지 스타일 변경

kimhyunmook committed:
lastest

pers0n4 committed:
Merge pull request #8 from kimhyunmook/basic-김현묵-sprint1,2

[김현묵] sprint 1+2

heonq committed:
Merge branch 'main' into basic-함헌규-sprint2
